cyral/resource_cyral_integration_idp_saml_draft.go

```diff
@@ -0,0 +1,280 @@
+				Description: "Identity provider type. The value provided can be used as a filter when retrieving SAML integrations. See data source `cyral_integration_idp_saml`.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     "saml-provider",
+			},
+			"attributes": {
+				Description: "SAML Attribute names for the identity attributes required by the Cyral SP. Each attribute name MUST be at least 3 characters long.",
+				Type:        schema.TypeSet,
+				Optional:    true,
+				ForceNew:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"first_name": {
+							Description:  "The name of the attribute in the incoming SAML assertion containing the users first name (given name). Defaults to `firstName`.",
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							Default:      defaultUserAttributeFirstName,
+							ValidateFunc: validationStringLenAtLeast(3),
+						},
+						"last_name": {
+							Description:  "The name of the attribute in the incoming SAML assertion containing the users last name (family name). Defaults to `lastName`.",
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							Default:      defaultUserAttributeLastName,
+							ValidateFunc: validationStringLenAtLeast(3),
+						},
+						"email": {
+							Description:  "The name of the attribute in the incoming SAML assertion containing the users email address. Defaults to `email`.",
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							Default:      defaultUserAttributeEmail,
+							ValidateFunc: validationStringLenAtLeast(3),
+						},
+						"groups": {
+							Description:  "The name of the attribute in the incoming SAML assertion containing the users group membership in the IdP. Defaults to `memberOf`.",
+							Type:         schema.TypeString,
+							Optional:     true,
+							ForceNew:     true,
+							Default:      defaultUserAttributeGroups,
+							ValidateFunc: validationStringLenAtLeast(3),
+						},
+					},
+				},
+			},
+			"sp_metadata": {
+				Description: "The SP Metadata document describing the Cyral service provider for this integration.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"id": {
+				Description: "ID of this resource in the Cyral environment.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			// toggle_recreation is used to recreate the SAML draft
+			// in case a SAML integration that points to it is
+			// deleted, but the SAML draft remains dangling in the
+			// Terraform configuration. In this case, the object
+			// wouldn't exist in the API, only in the Terraform
+			// configuration. To help the user get out of this
+			// situation, we force its recreation when he attempts
+			// `terraform apply` again.
+			"toggle_recreation": {
```

Comment:
I dont remember the conclusion we reached regarding this field, but I still think it's unnecessary and that we should consider removing it, as it adds just redundancy and makes the code more complex. Feel free to decide that though, as I dont want to block the PR with this.

Reply:
Solved, thanks for the heads up!